File: iOS-基础知识点/1_Block原理/block中的weak和storng02/ViewController.m
//
//  ViewController.m
//  block中的weak和storng02
//
//  Created by 林域 on 2021/8/29.
//

#import "ViewController.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
}


-(void)touchesBegan:(NSSet<UITouch *> *)touches withEvent:(UIEvent *)event{
    
    [self presentViewController:[NSClassFromString(@"TestViewController") new] animated:YES
                     completion:nil];
}

@end

File: iOS-UI知识点/UIView:Controller生命周期/UIView生命周期/ViewController.swift
//
//  ViewController.swift
//  UIView生命周期
//
//  Created by MAC on 2022/8/30.
//

import UIKit

class ViewController: UIViewController {
    
    override func viewDidLoad() {
        super.viewDidLoad()
    }
    
    /* 使用storyboard时，控制器的构造器为init(coder:)。
       该构造器为必须构造器required，因此重写其他构造器时，都必须重写该构造器。
     */
    required init?(coder: NSCoder) {
        super.init(coder: coder)
        print("使用storyboard时 构造器为：" + #function)
    }
    
    //使用storyboard时才会调用
    //使用该方法时，必须保证outlet和action连接完成
    //必须调用super避免不必要的错误
    override func awakeFromNib() {
        super.awakeFromNib()
        print(#function)
    }
    
    override func touchesBegan(_ touches: Set<UITouch>, with event: UIEvent?) {
        if let vc = TestViewController.init(nibName: "TestViewController", bundle: nil) as? TestViewController {
            self.navigationController?.pushViewController(vc, animated: true)
        }
    }
}

